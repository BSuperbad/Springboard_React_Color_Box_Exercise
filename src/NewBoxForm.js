import React, {useState} from "react";

const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = {
        color: "",
        width: "",
        height:""
    }

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addBox({...formData});
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color">Color: </label>
            <input 
            id="color"
            type="text" 
            name="color"
            placeholder="Box Color"
            value={formData.color}
            onChange={handleChange}
            />
            <label htmlFor="width">Width: {formData.width}</label>
            <input 
            id="width"
            type="range"
            name="width"
            min="100"
            max="500"
            value ={formData.width}
            onChange={handleChange}
            />
            <label htmlFor="height">Height: {formData.height}</label>
            <input 
            id="height"
            type="range"
            name="height"
            min="100"
            max="500"
            value ={formData.height}
            onChange={handleChange}
            />

            <button type="submit">Add Box</button>

        </form>
    )
}

export default NewBoxForm;