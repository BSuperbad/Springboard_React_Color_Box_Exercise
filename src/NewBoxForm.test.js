import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it("renders New Box Form without crashing", function () {
    render(<NewBoxForm />);
});

it("matches New Box Form snapshot", function() {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});

  it("should call addBox function on form submission", () => {
    const addBoxMock = jest.fn();
    const { getByLabelText, getByText } = render(<NewBoxForm addBox={addBoxMock} />);

    fireEvent.change(getByLabelText("Color:"), { target: { value: "blue" } });
    fireEvent.change(getByLabelText("Width:"), { target: { value: "250" } });
    fireEvent.change(getByLabelText("Height:"), { target: { value: "400" } });

    fireEvent.click(getByText("Add Box"));

    expect(addBoxMock).toHaveBeenCalledWith({ color: "blue", width: "250", height: "400" });
  });



