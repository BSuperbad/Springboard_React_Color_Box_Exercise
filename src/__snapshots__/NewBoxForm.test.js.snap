// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Box snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Color: 
    </label>
    <input
      id="color"
      name="color"
      placeholder="Box Color"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      id="width"
      max="500"
      min="100"
      name="width"
      type="range"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      id="height"
      max="500"
      min="100"
      name="height"
      type="range"
      value=""
    />
    <button
      type="submit"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;

exports[`matches New Box Form snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Color: 
    </label>
    <input
      id="color"
      name="color"
      placeholder="Box Color"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      id="width"
      max="500"
      min="100"
      name="width"
      type="range"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      id="height"
      max="500"
      min="100"
      name="height"
      type="range"
      value=""
    />
    <button
      type="submit"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
